internal/monitors/device-updater/device_updater.go

```diff
@@ -76,8 +85,75 @@ func (d *DeviceUpdater) updateDeviceSpecAccordingToFleetTemplate(log logrus.Fiel
 	device.Generation = &targetGeneration
 	device.Spec = model.MakeJSONField(fleet.Spec.Data.Template.Spec)
 	log.Infof("Updating device %s to generation %d", device.Name, *device.Generation)
-	err := d.devStore.UpdateDeviceInternal(device)
-	if err != nil {
-		log.Errorf("Failed to update target generation for device %s (fleet %s): %w", device.Name, fleet.Name, err)
+}
+
+func (d *DeviceUpdater) updateDeviceHeartbeatConditions(log logrus.FieldLogger, device *model.Device, fleet *model.Fleet) {
+	if device.Status.Data.UpdatedAt == nil {
```

Comment:
can we add a note why it's okay to bail here?

Reply:
I actually already check this above in line 55, removing it

Reply:
Sure